espnet2/enh/espnet_model.py

```diff
@@ -169,14 +172,38 @@ def forward_enhance(
         additional: Optional[Dict] = None,
     ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
         feature_mix, flens = self.encoder(speech_mix, speech_lengths)
-        feature_pre, flens, others = self.separator(feature_mix, flens, additional)
+        if self.mask_module is None:
```

Comment:
I’m actually not sure what is the mask module.
Can you explain it to me?

Reply:
Let me clarify what `mask_module` and `tcn_separator` are here.
`mask_module` corresponds to the final 1x1 conv block and nonlinear activation function in `tcn_separator` for estimating the masks. On the other hand, `tcn_separator_nomask` is `tcn_separator` without the mask estimation part.
I split `tcn_separator` into `tcn_separator_nomask` and `mask_module` so that:
- the intermediate bottleneck features (output of `tcn_separator_nomask`) can be fed to the diarization branch (the original `tcn_separator` does not output such features)
- masks corresponding to the given number of speakers are estimated (the number of speakers is fixed in the `tcn_separator`)

Reply:
Got it!
Thanks for the clarification.
I think you may already do that, but it would be great to leave such comments in the source code.

Reply:
Thanks! Will do.